command/agent/http.go

```diff
@@ -745,6 +748,7 @@ func (s *HTTPServer) wrap(handler func(resp http.ResponseWriter, req *http.Reque
 
 			resp.WriteHeader(code)
 			resp.Write([]byte(errMsg))
+			resp.Header().Set(contentTypeHeader, plainContentType)
```

Comment:
```suggestion
			resp.Header().Set(contentTypeHeader, plainContentType)
			resp.WriteHeader(code)
			resp.Write([]byte(errMsg))
```

I think setting the header needs to come before the other 2 response calls.

Reply:
oh, does it matter? I didn't realize. I'll update the rest of occurrences.

Reply:
setting the headers first ensures that the proper metadata / content-type can be sent back which reduces the risk possible wrong encoding or other response manipulation

Reply:
ah `WriteHeader` writes the header immediately to the response object 🤦‍♂️ 
"it's all coming back to me now"

Reply:
fixed in 7410b7312efcb2eac422615d8be138cd2ca3606d